tutorials/shaders/shader_reference/shading_language.rst

```diff
@@ -637,13 +637,25 @@ Example below:
         result = a + b;
     }
 
-.. note::
+Starting off Godot 4.4 you can now define multiple functions with the same name but different
+arguments (**overloading**). An implicit constant conversion (e.g. `1` -> `1.0`) will be disabled
+on function call for such cases to prevent ambiguity.
+
+.. code-block:: glsl
+
+    vec3 foo(int t) {
+        return vec3(1, 0, 0); // Red color.
+    }
+
+    vec3 foo(float t) {
+        return vec3(0, 1, 0); // Green color.
+    }
+
+    void foo() {
+        vec3 red = foo(1);
+        vec3 green = foo(1.0);
+    }
```

Comment:
```suggestion
    vec3 get_color(int t) {
        return vec3(1, 0, 0); // Red color.
    }

    vec3 get_color(float t) {
        return vec3(0, 1, 0); // Green color.
    }

    void fragment() {
        vec3 red = get_color(1);
        vec3 green = get_color(1.0);
    }
```
Nitpick, but avoid `foo()` and `bar()` style examples if possible. And in this case, use `fragment()` instead of `foo()` as the calling function to avoid confusion.